### PR TITLE
Refactor TimesBlock projection accumulation

### PR DESCRIPTION
**Pull request by ShinDongWoon**
## Summary
- refactor `TimesBlock.forward` to stream each path through its corresponding projection slice and accumulate in place, avoiding concatenation of intermediate feature maps.

## Testing
- `python - <<'PY'
import torch
import sys
from pathlib import Path
sys.path.append(str(Path(__file__).resolve().parents[0] / 'src'))

from timesnet_forecast.models.timesnet import TimesBlock

channels = 4
block = TimesBlock(channels=channels, kernel_set=[3,5], dropout=0.0, activation="gelu")
x = torch.randn(2, channels, 8, 6, requires_grad=True)

out = block(x)
loss = out.sum()
loss.backward()

print('x grad:', x.grad is not None)
print('proj weight grad shape:', block.proj.weight.grad.shape)
print('proj weight grad finite:', torch.isfinite(block.proj.weight.grad).all().item())
PY`
- `pytest tests/test_timesnet_forward.py`
- `pytest tests/test_dummy_training.py::test_dummy_training_smape_wsmape` *(fails: SMAPE 0.1016 > 0.1; training completes without OOM spikes)*

------
https://chatgpt.com/codex/tasks/task_e_68ccb7b779988328a38ba1bf6b79ad25

### CODE REVIEW

**Comment by ShinDongWoon** [comment]
@codex review
